include/RAJA/policy/cuda/kernel/CudaKernel.hpp

```diff
@@ -173,70 +173,61 @@ struct StatementExecutor<statement::CudaKernelExt<LaunchConfig,
     cudaStream_t stream = 0;
 
 
-    //
     // Compute the MAX physical kernel dimensions
-    //
-
     using data_t = camp::decay<Data>;
     LaunchDim max_physical = LaunchConfig::calc_max_physical(
         CudaKernelLauncher<StatementList<EnclosedStmts...>, data_t>, shmem);
 
-    //    printf("Physical limits: %d blocks, %d threads\n",
-    //        (int)max_physical.blocks, (int)max_physical.threads);
-
 
-    //
-    // Compute the Logical kernel dimensions
-    //
+    printf("Max: blocks=%d, threads=%d\n", (int)max_physical.blocks, (int)max_physical.threads);
 
     // Privatize the LoopData, using make_launch_body to setup reductions
     auto cuda_data = RAJA::cuda::make_launch_body(
         max_physical.blocks, max_physical.threads, shmem, stream, data);
-    //    printf("Data size=%d\n", (int)sizeof(cuda_data));
 
 
-    // Compute logical dimensions
-    using SegmentTuple = decltype(data.segment_tuple);
-
     // Instantiate an executor object
     using executor_t = cuda_statement_list_executor_t<stmt_list_t, data_t>;
     executor_t executor;
 
+
     // Compute logical dimensions
     LaunchDim logical_dims = executor.calculateDimensions(data, max_physical);
 
 
-    //    printf("Logical dims: %d blocks, %d threads\n",
-    //        (int)logical_dims.blocks, (int)logical_dims.threads);
+    printf("Logical: blocks=%d, threads=%d\n", (int)logical_dims.blocks, (int)logical_dims.threads);
 
-
-    //
     // Compute the actual physical kernel dimensions
-    //
-
     LaunchDim launch_dims;
     launch_dims.blocks = std::min(max_physical.blocks, logical_dims.blocks);
     launch_dims.threads = std::min(max_physical.threads, logical_dims.threads);
 
-    //    printf("Launch dims: %d blocks, %d threads\n",
-    //        (int)launch_dims.blocks, (int)launch_dims.threads);
+    printf("Launch: blocks=%d, threads=%d\n", (int)launch_dims.blocks, (int)launch_dims.threads);
+
+    // Only launch kernel if we have something to iterate over
+    bool at_least_one_iter = launch_dims.blocks > 0 || launch_dims.threads > 0;
+    bool is_degenerate =     launch_dims.blocks < 0 || launch_dims.threads < 0;
```

Comment:
Do you think the code will be offended if it realizes you are calling it degenerate?

Reply:
i hope so :)